docs/DeveloperGuide.md

```diff
@@ -1,38 +1,190 @@
+
+The `PC` class has methods to set and get the components of a `PC` object, as well as methods to set and get the name, isPrebuilt, and budget attributes.
+
+Overall, this class diagram provides a high-level overview of the components that make up a computer system and their relationships to each other.
+
+### Command class
+#### `list [COMPONENT]` Command (builder mode)
+The `list [COMPONENT]` command prints out a formatted list of all available components of type `[COMPONENT]`.
+
+When the user inputs a command of the form `list [COMPONENT]` in builder mode,
+it is parsed by the `Parser.parseBuilderCommand` method which recognizes the first word as
+`list` and creates a new `BuilderListComponentCommand` object while passing as argument the component
+which was part of the command.
+
+The `BuilderListComponentCommand.executeCommand()` method is then executed as part of the loop
+inside `BigPP.runLoopUntilExit()`.
+
+It first verifies:
+- That the component argument is not empty. If it is then it throws a `BuilderMissingComponentException()`
+- That the component given by the user exists in the database. If it doesn't then it throws a
+  `BuilderMissingComponentException()`
+
+It then uses the `stringToComponentListMap` HashMap from data storage to get the ComponentList associated
+with the component string input by the user (e.g. `cpu`). The `stringToComponentListMap` HashMap is initialized and
+filled when the program is first run using `DataStorage.initStringToComponentListMap()` and maps component names
+(e.g. `cpu`) to their corresponding `ComponentList` object (e.g. an object of type `CPUList`).
+
+The `ComponentList` class inherits from `ArrayList` but additionally implements the `getListString()` method which
+returns a formatted list of all the components in the `ComponentList` object.
+
+The `ComponentList` objects for each component are instantiated and populated in the `DataStorage.loadAll` method
+by loading the data in `resources/COMPONENT_NAME.json` using the `GSON` library.
+
+The program then checks if the user provided any flags to filter the output of the component command (e.g. by name,
+brand, price, etc.). If they did, then the program filters out the components not meeting the provided flags and
+creates a `ComponentList` containing all the components which do meet the provided flags.
+
+The program then uses the `ComponentList.getListString()` method to get a formatted list of the components
+of the desired type, which it then outputs to the user.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this command can be
+found below:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![List Component Command](./uml-pictures/listComponentCommand.png)
```

Comment:
Shouldn't User & BigPP need activation bar? Also, it might be better to "hide foot-box."

![image](https://user-images.githubusercontent.com/109506859/228428504-f65955f2-903e-40db-afe8-9705edf45784.png)